Disable Post `Create` and `Update` by default

pfefferle opened the pull request:
Introduces a new 'activitypub_create_posts' option to control whether posts can be created or updated via ActivityPub. Updates Posts::add and Posts::update to require a user ID and store it in post meta. Adjusts handlers and tests to accommodate the new parameter and option.

## Proposed changes:
- Adds a new 'activitypub_create_posts' option that defaults to false, requiring explicit opt-in for post creation/updates
- Updates Posts::add and Posts::update methods to require and store user ID in post meta
- Modifies handlers to check the new option before allowing post operations

### Other information:

- [x] Have you written new tests for your changes, if applicable?

## Testing instructions:
<!-- If you were reviewing this PR, how would you like the instructions to be presented? -->
<!-- Please include detailed testing steps, explaining how to test your change. -->
<!-- Bear in mind that context you working on is not obvious for everyone.  -->
<!-- Adding "simple" configuration steps will help reviewers to get to your PR as quickly as possible. -->
<!-- "Before / After" screenshots can also be very helpful when the change is visual. -->

* Go to '..'
*

### Changelog entry

<!-- You can optionally choose to enter a changelog entry by checking the box below and supplying data. -->
<!-- It will trigger a GitHub workflow that will create and push the entry into the branch. -->

<!-- Due to org permissions, the job may fail for PRs crated from a fork under GitHub organizations. -->
<!-- In this case, you can create entry manually with `composer changelog:add` and push it into the branch. -->

<!-- If no changelog entry is required for this PR, please add the "Skip Changelog" label. -->

-   [ ] Automatically create a changelog entry from the details below.

<details>

<summary>Changelog Entry Details</summary>

#### Significance

<!-- Choose only one -->

-   [ ] Patch
-   [ ] Minor
-   [ ] Major

#### Type

<!-- Choose only one -->

-   [ ] Added - for new features
-   [ ] Changed - for changes in existing functionality
-   [ ] Deprecated - for soon-to-be removed features
-   [ ] Removed - for now removed features
-   [ ] Fixed - for any bug fixes
-   [ ] Security - in case of vulnerabilities

#### Message

<!-- Add a changelog message here -->

</details>

pfefferle committed:
Add support for creating posts via ActivityPub option

Introduces a new 'activitypub_create_posts' option to control whether posts can be created or updated via ActivityPub. Updates Posts::add and Posts::update to require a user ID and store it in post meta. Adjusts handlers and tests to accommodate the new parameter and option.

pfefferle committed:
Add option setup and cleanup in create and update tests

The tests for create and update handlers now explicitly set and delete the 'activitypub_create_posts' option before and after execution. This ensures a consistent test environment and prevents side effects between tests.

pfefferle committed:
Move user ID meta addition after post insertion

The '_activitypub_user_id' meta is now added using add_post_meta after the post is inserted, rather than via the meta_input argument in wp_insert_post. This change ensures consistent handling of post meta and aligns with how other meta fields are added.

pfefferle committed:
Merge branch 'trunk' into posts-tweaks

pfefferle committed:
Merge branch 'trunk' into posts-tweaks

pfefferle committed:
Merge branch 'trunk' into posts-tweaks

obenland commented on includes/collection/class-posts.php:
```diff
@@ -114,6 +117,11 @@ public static function update( $activity ) {
 
+		$post_meta = \get_post_meta( $post_id, '_activitypub_user_id', false );
+		if ( \is_array( $post_meta ) && ! \in_array( (string) $user_id, $post_meta, true ) ) {
+			\add_post_meta( $post_id, '_activitypub_user_id', $user_id );
```
Is this to determine who to serve that content to? Maybe we should just add all user_ids that follow that actor?

When someone adds a new follow, we should probably add that user_id to that list as well, so they have access to existing posts?

pfefferle replied:
I think we should process every inbox request separately, because we have to get the visibility right. Only because it ends up in the Posts steam of one user does not mean that is meant for every other follower.

pfefferle replied:
I have an idea for the shared inbox that might help with that.

pfefferle replied:
this would require a major update though, because it is a breaking change 😱

pfefferle replied:
Let's not do it as part of this PR, because it brings a new layer of complexity we have to solve in general!

pfefferle committed:
Merge branch 'trunk' into posts-tweaks

pfefferle committed:
Add _activitypub_user_id post meta registration

Registers the '_activitypub_user_id' post meta for the custom post type, allowing storage of the local user ID that received the activity. This meta field is defined as an integer, single value, with a description and sanitization callback.

pfefferle committed:
Merge branch 'trunk' into posts-tweaks
